ShoppingList/src/test/java/edu/agh/eaiib/CommandParserTest.java

```diff
@@ -0,0 +1,41 @@
+package edu.agh.eaiib;
+
+import edu.agh.eaiib.model.Product;
+import edu.agh.eaiib.service.InMemoryProductListRepository;
+import edu.agh.eaiib.service.ProductListService;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class CommandParserTest {
+
+    private CommandParser testObject;
+    private ProductListService service;
+
+    @Before
+    public void setUp() {
+        testObject = new CommandParser();
+        service = new ProductListService(new InMemoryProductListRepository());
+        CommandParser.service = service;
+    }
+
+    @Test
+    public void testAddProduct() {
+        testObject.parse("add 3 apple");
```

Comment:
You can add `given when then` sections for better readability

Reply:
Yeah, I usually do that, just a little lazy.

Reply:
🤣